open_lm/main.py

```diff
@@ -467,17 +469,49 @@ def main(args):
     if args.train_data or (args.dataset_metadata is not None):
         named_parameters = list(model.named_parameters())
         no_decay_params = []  # to be potentially used later
-        params = [p for n, p in named_parameters if p.requires_grad]
 
-        optimizer = optim.AdamW(
-            [
-                {"params": no_decay_params, "weight_decay": 0.0},
-                {"params": params, "weight_decay": args.wd},
-            ],
-            lr=args.lr,
-            betas=(args.beta1, args.beta2),
-            eps=args.eps,
-        )
+        if args.mup_base_fan_in:
```

Comment:
maybe this should be base width and not base fan-in, because fan-in is not fixed throughout the network (fan-in = # of inputs, so w3 has a different fan-in than in_proj, out_proj, w12)

Reply:
sweet! changed!